Gordon360/Controllers/RecIM/TeamsController.cs

```diff
@@ -0,0 +1,282 @@
+            
+        }
+
+        /// <summary>
+        /// Add a participant to a team
+        /// </summary>
+        /// <param name="teamID"></param>
+        /// <param name="participant">Default Role Value value 3 (Member)</param>
+        /// <returns></returns>
+        [HttpPost]
+        [Route("{teamID}/participants")]
+        [StateYourBusiness(operation = Operation.UPDATE, resource = Resource.RECIM_TEAM)]
+        public async Task<ActionResult<ParticipantTeamViewModel>> AddParticipantToTeam(int teamID, ParticipantTeamUploadViewModel participant)
+        {
+            var inviterUsername = AuthUtils.GetUsername(User);
+            var activityID = _teamService.GetTeamActivityID(teamID);
+            if (!_teamService.HasUserJoined(activityID, participant.Username) || _participantService.IsAdmin(inviterUsername))
+            {
+                var participantTeam = await _teamService.AddParticipantToTeamAsync(teamID, participant, inviterUsername);
+                return CreatedAtAction("AddParticipantToTeam", participantTeam);
+            }
+            else
+                return UnprocessableEntity($"Participant {participant.Username} already is a part of a team in this activity");
+        }
+
+        /// <summary>
+        /// Updates Participant role in a team
+        /// </summary>
+        /// <param name="teamID"></param>
+        /// <param name="participant">Default Role Value value 3 (Member)</param>
+        /// <returns></returns>
+        [HttpPatch]
+        [Route("{teamID}/participants")]
+        [StateYourBusiness(operation = Operation.UPDATE, resource = Resource.RECIM_TEAM)]
+        public async Task<ActionResult<ParticipantTeamViewModel>> UpdateParticipantTeam(int teamID, ParticipantTeamUploadViewModel participant)
+        {
+            
+                participant.RoleTypeID = participant.RoleTypeID ?? 3;
+                var participantTeam = await _teamService.AddParticipantToTeamAsync(teamID, participant);
+                return CreatedAtAction("AddParticipantToTeam", participantTeam);
+            
+        }
+
+        /// <summary>
+        /// Removes specified user from a team
+        /// </summary>
+        /// <param name="teamID"></param>
+        /// <param name="username"></param>
+        /// <returns></returns>
+        [HttpDelete]
+        [Route("{teamID}/participants")]
+        public async Task<ActionResult> DeleteTeamParticipant(int teamID, string username)
+        {
+            var user_name = AuthUtils.GetUsername(User);
+            var participantTeam = _teamService.GetParticipantTeam(teamID, username);
+            if (participantTeam is null)
+                return NotFound("The user is not part of the team.");
+            if (user_name != participantTeam.ParticipantUsername)
+                return Forbid($"You are not permitted to reject invitations for another participant.");
+
+            await _teamService.DeleteParticipantTeamAsync(teamID, username);
```

Comment:
I think this specific function gets used in handling automated deletes as well as captain/co-captain deletes. In my mind I do think there are other functions that need to be able to delete regardless of whether or not they're actually the targeted participant. However, those other functions are all locked behind routes with state your business. I'll see if I can do a work around and add a non intrusive check in the function, but that would be based more on preference. 

tldr; if you really want me to, I can try to add a service check, otherwise it's not "freely" accessible